<launch>
  <arg name="nodelet_manager_launched" default="0" />
  <arg name="serial" default="0" />
  <arg name="side" default="left" />
  <arg name="enable_trigger" default="true"/>
  
  <!-- nodelet manager process -->
  <node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" machine="autorally-master" output="screen" args="manager" unless="$(arg nodelet_manager_launched)" />
  
  <param name="/camera/image_raw/compressed/jpeg_quality" type="int" value="10"/>

  <!-- camera driver nodelet -->
  <node pkg="nodelet" type="nodelet" name="$(arg side)_camera_nodelet"
        args="load spinnaker_camera_driver/SpinnakerCameraNodelet camera_nodelet_manager" >

    <param name="frame_id" value="camera"/>
    <param name="serial" value="$(arg serial)"/>
    <param name="camera_info_url" value="file://$(env AR_CONFIG_PATH)/camera_calibration_$(arg serial).yaml" />

    <remap from="image_raw" to="$(arg side)_camera/image_raw" />
    <remap from="set_camera_info" to="$(arg side)_camera/set_camera_info"/>
    <remap from="image" to="$(arg side)_camera/image_diagnostic"/>
  </node>

  <!-- load parameters for camera nodelet -->
  <node pkg="dynamic_reconfigure" type="dynparam" name="dynamic_configurator"
        args="load spinnaker_camera_nodelet $(dirname)/flir_test_dynparams.yaml">
  </node>

  <!-- No debayering is needed; using this mostly for republishing on correct topics -->
  <node pkg="nodelet" type="nodelet" name="$(arg side)_image_proc_debayer" args="load image_proc/debayer camera_nodelet_manager">
    <remap from="image_raw" to="$(arg side)_camera/image_raw" />
    <remap from="image_mono" to="$(arg side)_camera/image_mono" />
    <remap from="image_color" to="$(arg side)_camera/image_color" />
    <param name="debayer" value="0"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="$(arg side)_image_proc_rect_mono" args="load image_proc/rectify camera_nodelet_manager">
    <remap from="image_mono" to="$(arg side)_camera/image_mono" />
    <remap from="image_rect" to="$(arg side)_camera/image_mono_rect" />
  </node>
  <node pkg="nodelet" type="nodelet" name="$(arg side)_image_proc_rect_color" args="load image_proc/rectify camera_nodelet_manager">
    <remap from="image_mono" to="$(arg side)_camera/image_color" />
    <remap from="image_rect" to="$(arg side)_camera/image_color_rect" />
  </node>

  <node pkg="nodelet" type="nodelet" name="$(arg side)_ImageRepublisher" args="load autorally_core/ImageRepublisher camera_nodelet_manager" output="screen" machine="autorally-master" >
    <param name="fps" value="10" />
    <param name="resizeHeight" value="480" />
    <remap from="camera/image_raw" to="$(arg side)_camera/image_color" />
    <remap from="camera/image_display" to="$(arg side)_camera/image_display" />
  </node>
  
  <!-- <include file="$(find autorally_core)/launch/camera_auto_balance.launch">
    <arg name="serial" value="$(arg serial)" />
    <arg name="side" value="$(arg side)" />
  </include> -->

</launch>
